### ISSUE TITLE
Wrong memory order constraint in caml_modify

### ISSUE THREAD

**Issue by OlivierNicole** [ocaml/ocaml#10886]
This issue is a duplicate of review comment https://github.com/ocaml/ocaml/pull/10831#discussion_r781324239.

At https://github.com/ocaml/ocaml/blob/1546e1f86f87485637719c86d92f32df9292997f/runtime/memory.c#L156, it should be `memory_order_relaxed` instead of `memory_order_release`.

**Comment by gadmm** [comment]
This has been clarified by Stephen today. A release is needed between initialisation and publication of the value, so that the initialisation of the value is seen by other threads. The straightforward thing to do is to have a release in `caml_alloc` but it is less expensive to have this ordering in the first `caml_modify` that publishes the value. This is mentioned in the comment but was not entirely clear to me, I propose to clarify the comment.

**Comment by OlivierNicole** [comment]
I don't see it mentioned in the current comment.

**Comment by gadmm** [comment]
My point :)
```
Using release stores for all writes also ensures publication safety
for newly-allocated objects, and isn't necessary for initialising
writes.
```

**Comment by OlivierNicole** [comment]
Right, you pointed it yourself in your comment https://github.com/ocaml/ocaml/pull/10831#discussion_r781322577 that this sentence is misleading. Well, maybe I should let you make the change :smile:

**Comment by kayceesrk** [comment]
I believe that this specific memory ordering was extensively discussed in https://github.com/ocaml/ocaml/pull/10972. Can this issue be closed?

**Comment by gadmm** [comment]
Yes

**Comment by kayceesrk** [comment]
Thanks @gadmm. Closing this one. Fixed in #10972.
